course/models.py

```diff
@@ -114,3 +115,62 @@ class Review(models.Model):
     content = models.TextField(blank=True)
     date = models.DateField(auto_now_add=True)
     objects = ReviewManager()
+
+
+class StudentCoursesManager(models.Manager):
+
+    def get_teacher_pending(self, id: int):
+        return TeacherCourses.objects.get_teacher_courses(id).values("course_id").intersection(
+            self.filter(status=Status.PENDING).values("teacher_course_id"))
+
+    def get_student_pending(self, id: int):
+        return self.filter(student_id=id, status=Status.PENDING)
+
+    def get_student_courses(self, id: int):
+        return self.filter(student_id=id)
+
+
+class Status(models.TextChoices):
+    PENDING = 'Pending'
+    CONFIRMED = 'Confirmed'
+
+
+class StudentCourses(models.Model):
+    student_course_id = models.BigAutoField(primary_key=True)
+    student_id = models.ForeignKey(Profile, on_delete=models.CASCADE)
+    teacher_course_id = models.ForeignKey(TeacherCourses, on_delete=models.CASCADE)
+    request_date = models.DateTimeField(auto_now_add=True)
+    objects = StudentCoursesManager()
+
+    status = models.CharField(
+        max_length=30,
+        choices=Status.choices,
+        default=Status.PENDING,
+        editable=False
+    )
+
+    def change_to_confirmed(self):
+        self.status = Status.CONFIRMED.value
+        self.save()
+
+    def Teacher_course_id(self):
+        return str(self.teacher_course_id.course_id)
+
+    def clean(self):
+        if self.teacher_course_id.teacher_id == self.student_id:
+            raise ValidationError({'teacher_course_id': "Can't create a request to your own course"})
+        for course in StudentCourses.objects.get_student_courses(self.student_id):
+            if course.teacher_course_id == self.teacher_course_id:
```

Comment:
Maybe you can use [UniqueConstraint](https://docs.djangoproject.com/en/4.2/ref/models/constraints/#uniqueconstraint) for doing this?